exercises/palindrome-products/test/Tests.hs

```diff
@@ -1,73 +1,36 @@
-    let (largest, factors) = largestPalindrome 10 99
-    (9009 :: Int) @=? largest
-    [(91, 99)] @=? norm factors
-  , testCase "smallest palindrome from double digit factors" $ do
-    let (smallest, factors) = smallestPalindrome 10 99
-    (121 :: Int) @=? smallest
-    [(11, 11)] @=? norm factors
-  , testCase "largest palindrome from triple digit factors" $ do
-    let (largest, factors) = largestPalindrome 100 999
-    (906609 :: Int) @=? largest
-    [(913, 993)] @=? norm factors
-  , testCase "smallest palindrome from triple digit factors" $ do
-    let (smallest, factors) = smallestPalindrome 100 999
-    (10201 :: Int) @=? smallest
-    [(101, 101)] @=? norm factors
-  , testCase "largest palindrome from four digit factors" $ do
-    let (largest, factors) = largestPalindrome 1000 9999
-    (99000099 :: Int) @=? largest
-    [(9901, 9999)] @=? norm factors
-  , testCase "smallest palindrome from four digit factors" $ do
-    let (smallest, factors) = smallestPalindrome 1000 9999
-    (1002001 :: Int) @=? smallest
-    [(1001,1001)] @=? norm factors
-  , testCase "largest palindrome from five digit factors" $ do
-    let (largest, factors) = largestPalindrome 10000 99999
-    (9966006699 :: Integer) @=? largest
-    [(99681, 99979)] @=? norm factors
-  , testCase "smallest palindrome from five digit factors" $ do
-    let (smallest, factors) = smallestPalindrome 10000 99999
-    (100020001 :: Integer) @=? smallest
-    [(10001,10001)] @=? norm factors
-  ]
+main = hspecWith defaultConfig {configFastFail = True} specs
+
+specs :: Spec
+specs = describe "palindrome-products" $ for_ cases test
+  where
+    test (desc, minFactor, maxFactor, sPal, sPalFactors, lPal, lPalFactors) =
+      describe desc $ do
+        let sortPair (a, b)  = if a < b then (a, b) else (b, a)
+        let normalize        = sort . nub . map sortPair
+        describe "smallesPalindrome" $ do
+          let (value, factors) = smallestPalindrome minFactor maxFactor
+          it "value"   $ value             `shouldBe` sPal
+          it "factors" $ normalize factors `shouldBe` sPalFactors
+        describe "largestPalindrome" $ do
+          let (value, factors) = largestPalindrome minFactor maxFactor
+          it "value"   $ value             `shouldBe` lPal
+          it "factors" $ normalize factors `shouldBe` lPalFactors
+
+    -- As of 2016-09-07, there was no reference file
+    -- for the test cases in `exercism/x-common`.
+
+    cases = [ ("palindromes from single digit factors",     1,     9,         1, [(    1,     1)],          9, [(1, 9), (3, 3)])
+            , ("palindromes from double digit factors",    10,    99,       121, [(   11,    11)],       9009, [(   91,    99)])
+            , ("palindromes from triple digit factors",   100,   999,     10201, [(  101,   101)],     906609, [(  913,   993)])
+            , ("palindrome from four digit factors"   ,  1000,  9999,   1002001, [( 1001,  1001)],   99000099, [( 9901,  9999)])
```

Comment:
Nice catch! Fixed! :smile: 

> Regarding the experiment to use HINTS.md: What result do you think you will use to decide whether to apply it to other exercises?

Right now we have hints as comments in the test suites and in the stub solutions.

The comments in the stub solutions proved to be a tragedy, because most of the users just leave the hints in the submitted code.

The comments in the test suite are not that bad, but I think that the users should get feedback primarily running the test suite, not inspecting it.

If the hints file works as expected, I think it would be probably a good idea to do the same for all exercises. But first I would like to check if it is correctly inserted in the `README.md` and if it doesn't have unexpected effects.

What do you think about it?
Any other track is using it?